src/napari_live_recording/control/__init__.py

```diff
@@ -77,139 +87,476 @@ def addCamera(self, cameraKey: str, camera: ICamera) -> str:
+        self.bufferWorkers[cameraKey] = self.recordToBuffer(cameraKey)
+        self.__isAcquiring[cameraKey] = True
+        self.bufferWorkers[cameraKey].start()
+
         return cameraKey
 
+    @thread_worker(worker_class=FunctionWorker, start_thread=False)
+    def recordToBuffer(self, cameraKey: str):
+        if self.__isAcquiring[cameraKey]:
+            while self.__isAcquiring[cameraKey]:
+                try:
+                    currentFrame = np.copy(
+                        self.deviceControllers[cameraKey].device.grabFrame()
+                    )
+                    self.deviceBuffers[cameraKey].addFrame(currentFrame)
+
+                    if self.recordingRunning[cameraKey]:
+                        self.recordingBuffers[cameraKey].addFrame(currentFrame)
+                        self.processingBuffers[cameraKey].addFrame(currentFrame)
+                except Exception as e:
+                    pass
+
     def changeCameraROI(self, cameraKey: str, newROI: ROI) -> None:
         self.deviceControllers[cameraKey].device.changeROI(newROI)
         self.deviceLiveBuffer[cameraKey] = np.zeros(shape=newROI.pixelSizes)
+        self.deviceBuffers[cameraKey].changeROI(newROI)
+        self.recordingBuffers[cameraKey].changeROI(newROI)
+        self.processingBuffers[cameraKey].changeROI(newROI)
+        self.processedBuffers[cameraKey].changeROI(newROI)
 
-    def deleteCamera(self, cameraKey: str) -> None:
+    def changeStackSize(self, newStackSize: int):
+        self.stackSize = newStackSize
+        for cameraKey in self.deviceControllers.keys():
+            self.recordingBuffers[cameraKey].changeStacksize(
+                newStacksize=self.stackSize
+            )
+            self.processingBuffers[cameraKey].changeStacksize(
+                newStacksize=self.stackSize
+            )
+            self.processedBuffers[cameraKey].changeStacksize(
+                newStacksize=self.stackSize
+            )
 
-        """Deletes a camera device. """
-        if self.isLive:
-            self.live(False)
+    def deleteCamera(self, cameraKey: str) -> None:
+        """Deletes a camera device."""
         try:
+            self.__isAcquiring[cameraKey] = False
+            _ = self.processingRunning.pop(cameraKey)
+            _ = self.recordingRunning.pop(cameraKey)
+            self.cameraDeleted.emit(False)
+
             self.deviceControllers[cameraKey].device.close()
             self.deviceControllers[cameraKey].thread.quit()
-            self.deviceControllers[cameraKey].thread.deleteLater()
             self.deviceControllers[cameraKey].device.deleteLater()
-            self.recordSignalCounter.maxCount -= 1
+            self.deviceControllers[cameraKey].thread.deleteLater()
+            self.bufferWorkers[cameraKey].quit()
+            self.deviceControllers[cameraKey].device.setAcquisitionStatus(False)
+
+            _ = self.deviceBuffers.pop(cameraKey)
```

Comment:
the `_ = `is not really necessary, you can remove it and just leave the call to `pop`